src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -349,6 +349,65 @@ BuildCommand BuildProject(string buildNumber)
             }
         }
 
+        [Fact]
+        public void It_includes_assembly_metadata()
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld")
+                .WithSource()
+                .WithTargetFramework("netstandard2.0")
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    project.Root.Add(
+                        new XElement(ns + "ItemGroup",
+                            new XElement(ns + "AssemblyMetadata",
+                                new XAttribute("Include", "MetadataKey"),
+                                new XAttribute("Value", "MetadataValue"))));
+                });
+
+            new RestoreCommand(Log, testAsset.TestRoot).Execute().Should().Pass();
```

Comment:
nit: I believe it's not usually necessary to explicitly restore before building anymore, is that correct @dsplaisted